### DISCUSSION TITLE
:ref not work with simple div or custom component

### DISCUSSION THREAD

**Discussion by JoFox911** [quasarframework/quasar#10543] [General - Components / Directives / etc] [answered]
Hi!
I'm using quasar v2.0.3 and @quasar/app. v3.0.4

when adding a ref to a common div, then I see it in the list of all refs
when adding a dynamic ref to an input, you can also see it in the list
but when adding a dynamic ref to div (:ref), it is not displayed in the list (a simple example even without variables below)
please tell me what could be the reason?

html code:
![image](https://user-images.githubusercontent.com/36991921/131624575-571969eb-8895-4724-bbd4-f8b5be69a85f.png)

this.$refs with debugger:
![image](https://user-images.githubusercontent.com/36991921/131624487-550362c2-f28f-42f5-99b6-9a6a070aabc7.png)

**Comment by ontwikkelfabriek** [comment, marked as answer]
@JoFox911 

Maybe this will help:
https://stackoverflow.com/questions/61155893/how-to-access-dynamic-ref-tagged-html-elements-in-vue-js-3
https://thewebdev.info/2021/05/22/how-to-add-refs-dynamically-in-vue-js/

**Reply by JoFox911** [reply]
after https://stackoverflow.com/questions/61155893/how-to-access-dynamic-ref-tagged-html-elements-in-vue-js-3
I replace :ref="item.id" with :ref="`notification-${item.id}`" and that helps))
for this https://stackoverflow.com/questions/61155893/how-to-access-dynamic-ref-tagged-html-elements-in-vue-js-3 thanks too)

**Reply by ontwikkelfabriek** [reply]
@JoFox911 

> I replace :ref="item.id" with :ref="notification-${item.id}" and that helps

Doesn't really make sense to me why this would solve it. Maybe one of the Vue3 gods could clarify this....
